trestle/core/rules_validator.py

```diff
@@ -0,0 +1,110 @@
+# -*- mode:python; coding:utf-8 -*-
+
+# Copyright (c) 2023 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Validate by confirming rule parameter values are consistent."""
+import logging
+import pathlib
+
+import trestle.oscal.catalog as cat
+from trestle.common.common_types import TopLevelOscalModel
+from trestle.common.common_types import TypeWithProps
+from trestle.common.list_utils import as_list, deep_get, deep_set, get_item_from_list
+from trestle.core.catalog.catalog_interface import CatalogInterface
+from trestle.core.profile_resolver import ProfileResolver
+from trestle.core.validator import Validator
+from trestle.oscal.ssp import SystemSecurityPlan
+
+logger = logging.getLogger(__name__)
+
+shared_params = {}
+
+
+class RulesValidator(Validator):
+    """Validator to confirm all rule parameter values are consistent."""
+
+    def add_shared_param(
+        self, item: TypeWithProps, control_id: str, cat_int: CatalogInterface, comp_name: str = ''
+    ) -> None:
+        """
+        Add a rule shared parameter to the rule shared parameters list.
+
+        args:
+            item: Generic item to iterate over parameters.
+            control_id: Current control id.
+            cat_int: Instance of catalog interface with controls catalog loaded.
+            comp_name: Component name to save.
+        """
+        for param in as_list(item.set_parameters if not isinstance(item, cat.Control) else item.params):
+            parameter_id = param.param_id if not isinstance(item, cat.Control) else param.id
```

Comment:
no need to check controls for params.  use the catalog interface to see if a given param_id is in the catalog.  If it is not, then assume it is a rule_param.

Reply:
done. resolved